core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Service.java

```diff
@@ -177,8 +178,11 @@ public Response finishOAuth2Challenge(String state, String code, URI callbackUri
                                                 tokenExpiration
                                                         .map(expiration -> Instant.now().plus(expiration))
                                                         .orElse(oauth2Response.getExpiration())),
-                                NonceCookie.delete())
-                        .build();
+                                NonceCookie.delete());
+                if (oauth2Response.getIdToken().isPresent()) {
+                    builder.cookie(OAuthIdTokenCookie.create(oauth2Response.getIdToken().get(), oauth2Response.getExpiration()));
```

Comment:
But in case of `OAuthWebUiCookie` we do wrap the token as JWT and pass the expiration time for that, which is independent of the ID token. So shouldn't we use them ?

Reply:
I see. How this should work for refresh tokens in that case? I think we will get `expiration` for the access token, let say, `1h` but since we will have refresh token we will be able to prolong the session for much longer than that. Using the same expiration for ID token cookie lifetime will mean that the this cookie will expire after `1h` and we won't be able to perform logout correctly after that.